### PR TITLE
shorten server names to avoid >63 character pod names

### LINKED ISSUE #248: shorten pod name for kubespawner
get this error on k8s: 

```
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"Pod \"jupyterhub-rok-2eroskar--89faaf422189e33ea4837b365cb84f89\" is invalid: [spec.volumes[0].name: Invalid value: \"jupyterhub-rok-2eroskar--89faaf422189e33ea4837b365cb84f89-git-repo\": must be no more than 63 characters
```

### PR DESCRIPTION
**Pull request by rokroskar**
pod names are now, e.g.: 

```
jupyter-cramakri--f590027
```

closes #248

### CODE REVIEW

File: charts/renku/values.yaml [hunk 1]
```diff
@@ -314,6 +314,10 @@ jupyterhub:
     extraConfig: |
       import os
 
+      # JupyterHub uses GITLAB_HOST which is a misnomer -- we use GITLAB_URL
+      # everywhere so set this mapping here
+      os.environ['GITLAB_HOST'] = os.getenv('GITLAB_URL')
```
Comment (jirikuncar) [inline charts/renku/values.yaml]: I hope we solve it quickly in https://github.com/jupyterhub/oauthenticator/issues/191.

**Comment by rokroskar** [comment]
@leafty another option would be to use `jupyter-<hash>` and include user and commit sha in the container/pod labels... that might be the safest/cleanest.

**Comment by leafty** [comment]
But then, when you list pods the name is not informative. It would be more simpler to lookup someone pod for me if it included the username (or part of).

**Comment by rokroskar** [comment]
ok then I would vote for trimming the username, e.g.

`jupyter` (7 chars) + `-<username>-` + `<hash>` (7 chars)

so `<username>` is trimmed to 46 characters to make sure it fits. I guess with 46 characters it should be easy to have unambiguous usernames :)

**Comment by rokroskar** [comment]
and actually, instead of using the hash, I will just use the first 7 digits of the commit sha.

**Comment by jirikuncar** [comment]
> 46 characters to make sure it fits

poor "Pablo Diego José Francisco de Paula Juan Nepomuceno María de los Remedios Cipriano de la Santísima Trinidad Ruiz y Picasso" but for him the "Pablo Picasso" would do 😉

**Comment by rokroskar** [comment]
In 47b491a:

The server name now includes namespace, project and commit-sha.
The pod and docker container names are of the form:

jupyter-<username>-<namespace>-<project>-<commit-sha>

"namespace" and "project" are limited to 10 characters,
"commit-sha" is limited to 7. This leaves 28 characters for
username to still yield a valid DNS name for k8s.

**Pull request merged**